File: test/fixtures-units/sample.go
package main

import "fmt"

type Server struct {
	name string
}

type Handler interface {
	Serve()
}

func (s *Server) Serve() {
	for i := 0; i < 3; i++ {
		fmt.Println(`raw { string`)
	}
	go func() {
		s.name = "x"
	}()
}

func main() {
	cfg := Config{Name: "a"}
	fn := func() {
		_ = cfg
	}
	fn()
	if err := run(); err != nil {
		return
	}
}

File: test/fixtures-units/sample.rs
use std::fmt;

/* block /* nested */ still comment { */
#[derive(Debug)]
pub struct Point<'a> {
    name: &'a str,
}

impl<'a> fmt::Display for Point<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let raw = r#"has " and { brace"#;
        let c = '{';
        write!(f, "{}", self.name)
    }
}

pub fn run<T>(items: Vec<T>) -> Result<(), String>
where
    T: Clone,
{
    let closure = |x: u32| {
        x + 1
    };
    match items.len() {
        0 => Err(String::new()),
        _ => Ok(()),
    }
}

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        assert_eq!(1, 1);
    }
}
